manifests/vanilla/vsphere-csi-driver.yaml

```diff
@@ -278,7 +278,7 @@ spec:
             - mountPath: /csi
               name: socket-dir
         - name: vsphere-csi-controller
-          image: us-central1-docker.pkg.dev/k8s-staging-images/csi-vsphere/ci/driver:latest
+          image: us-central1-docker.pkg.dev/k8s-staging-images/csi-vsphere/driver:latest
```

Comment:
Can we patch this file when we are cutting releases please?

https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/3.0/rn/vmware-vsphere-container-storage-plugin-30-release-notes/index.html#vSphere%20Container%20Storage%20Plug-in%203.1.x

The install guide refers to this manifest and the image registry should be registry.k8s.io.

Reply:
Yes this is for the main branch.

In the release branch we need to use release image registry. which is registry.k8s.io/csi-vsphere/* as per https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/3053#issuecomment-2358713717

so the process is after tag is cut and new image is pushed at the staging, we need to promote image to `registry.k8s.io/csi-vsphere/*` and then publish that path in the released YAML file.

Reply:
@upodroid is this understanding correct?

Reply:
@upodroid where do we see registry.k8s.io/csi-vsphere/*
i could not locate reference in the code base?

Reply:
Have a look at https://github.com/kubernetes/k8s.io/tree/main/registry.k8s.io

The image promoter will copy your image from the staging registry to registry.k8s.io and your consumers need pull the image from there.

Reply:
I think we can write a script to populate this file https://github.com/kubernetes/k8s.io/blob/main/registry.k8s.io/images/k8s-staging-csi-vsphere/images.yaml with images that show up in the staging repo. After that it will become available under `registry.k8s.io/csi-vsphere/*`.

Reply:
I wouldn't script that, instead of I would manually check the digests and insert it. The staging images expire so the script wouldn't be deterministic.

Also, are you updating this file to reference the correct registry? https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/v3.3.0/manifests/vanilla/vsphere-csi-driver.yaml#L280

Reply:
I recommend bundling the manifests that users will use in the releases. Look at kueue for example.

https://github.com/kubernetes-sigs/kueue/releases/tag/v0.8.1